### ISSUE TITLE
Incorporate the changes from @pahihu to support Java 11 and Java 17

### ISSUE THREAD

**Issue by 1997tn** [k-john-gough/gpcp#22]
I'm very glad to see @k-john-gough still around and accepting pull requests. @pahihu didn't fork from this repo, he populated his repo with the source code from the released zip archive. This make doing a pull request to incorporate his changes to this repo more difficult. I attempted to do only to realize I don't understand Java and know nothing about gpcp's internals. Anyone still around please consider making a pull request to incorporate @pahihu's changes. Thank you very much.

@proberon Please have a look at https://github.com/pahihu/gpcp-JVM, there are only 9 commits, but I don't understand anything to help. I don't want myself to create a mess for you all.

p/s: This commit should be skipped: https://github.com/pahihu/gpcp-JVM/commit/116c8c861d877f0480b1bbff6ddef3dc2d77237a as that document has nothing to do with gpcp: https://github.com/pahihu/gpcp-JVM/issues/2

**Comment by cfbsoftware** [comment]
> p/s: This commit should be skipped: [pahihu/gpcp-JVM@116c8c8](https://github.com/pahihu/gpcp-JVM/commit/116c8c861d877f0480b1bbff6ddef3dc2d77237a) as that document has nothing to do with gpcp: [pahihu/gpcp-JVM#2](https://github.com/pahihu/gpcp-JVM/issues/2)

I disagree. The document gives additional information about the latest changes that were made to the CP  *language* not a specific *implementation* of CP. On the contrary, it has much to do with GPCP as most, if not all, of those changes were implemented in GPCP.

**Comment by 1997tn** [comment]
> > p/s: This commit should be skipped: [pahihu/gpcp-JVM@116c8c8](https://github.com/pahihu/gpcp-JVM/commit/116c8c861d877f0480b1bbff6ddef3dc2d77237a) as that document has nothing to do with gpcp: [pahihu/gpcp-JVM#2](https://github.com/pahihu/gpcp-JVM/issues/2)
> 
> I disagree. The document gives additional information about the latest changes that were made to the CP _language_ not a specific _implementation_ of CP. On the contrary, it has much to do with GPCP as most, if not all, of those changes were implemented in GPCP.

I disagree with you. If the changes were already implemented, that document is OK. But the fact is they are not yet implemented, that document only causes confusion to the users.

**Comment by cfbsoftware** [comment]
> 
> I disagree with you. If the changes were already implemented, that document is OK. But the fact is they are not yet implemented, that document only causes confusion to the users.

Fact??? There seems to be some misunderstanding here. Which changes are you saying have not been implemented?

**Comment by 1997tn** [comment]
> > I disagree with you. If the changes were already implemented, that document is OK. But the fact is they are not yet implemented, that document only causes confusion to the users.
> 
> Fact??? There seems to be some misunderstanding here. Which changes are you saying have not been implemented?

See: https://github.com/pahihu/gpcp-JVM/issues/2

That document is for another compiler. It doesn't apply to gpcp!

**Comment by cfbsoftware** [comment]
> See: [pahihu/gpcp-JVM#2](https://github.com/pahihu/gpcp-JVM/issues/2)
> 
> That document is for another compiler. It doesn't apply to gpcp!

[pahihu/gpcp-JVM#2](https://github.com/pahihu/gpcp-JVM/issues/2) has now been corrected at my request, confirming that the document does in fact apply to GPCP.

**Comment by ghost** [comment]
@proberon Could you make a PR for this? It seems you are an enthusiast.

**Comment by pahihu** [comment]
Hi, 

Since my repo is not forked from the original, making a PR from it is a bit problematic.
I have attached patch files, which contains the changes. Apply them in numerical order.

Regards,
Andras Pahi

**Comment by k-john-gough** [comment]
Hi Chris

I have suddenly become aware of this discussion thread.
I don’t really understand what the issues are, so perhaps you could find time to give me a short summary?

Regards
John

**Comment by cfbsoftware** [comment]
Hi John,

 

The part of this issue that you referred to is already closed:

 

https://github.com/pahihu/gpcp-JVM/issues/2

 

I am not familiar with the JVM version of GPCP so am unable to help with that part of this issue,

 

Cheers,

Chris

**Comment by btreut** [comment]
as far as I  understand it, the patches created by Andras Pahi (aka pahihu) can be found here:
https://github.com/pahihu/gpcp-JVM/tree/main/patches
